### PR TITLE
First pass

### PR DESCRIPTION
**Pull request by maccman**

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+- config_type: visitor_data_service
+  master_label: clearbit
+  help_text: Build audiences based on Clearbit campaigns
+  data_fetch_method: custom_javascript
+  supported_fields:
```
Comment (maccman) [inline config.yaml]: Is this the correct format?

File: functions.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  fetchData: function() {
+    var params = {
+      authorization: options.apiKey
+    }
+
+    $.getJSON("https://reveal.clearbit.com/v1/optimizely", params, function(data) {
```
Comment (maccman) [inline functions.js]: This endpoint requires authentication. How do you pass an API key to this?

File: functions.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import requests
+
+from optimizely_platform import exceptions
+from optimizely_platform import objects
+
+
+def get_dynamic_audience_conditions(integration_settings):
```
Comment (maccman) [inline functions.py]: Our endpoint doesn't return any campaigns - just data. Is this correct?

File: config.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+- config_type: visitor_data_service
+  master_label: clearbit
```
Comment (aliabbasrizvi) [inline config.yaml]: I think this should be `Clearbit`. It should be what you want the name to appear like in the UI.

File: config.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  help_text: Build audiences based on Clearbit campaigns
+  data_fetch_method: custom_javascript
+  supported_fields:
+    - name: audience
+      display_name: Clearbit Audience
+      help_text: Classifies web traffic by defined categories
+      input_type: select
+      values:
```
Comment (aliabbasrizvi) [inline config.yaml]: To me the set of values that appear here appear to be parameters that you want to build the audience based on. In the current format, you will have a single dropdown with all these options. What you want is to write your conditions as follows:

```
supported_fields:
  - name: companyName
     display_name: Company Name
     help_text: Clearbit's name for the company identified.
     input_type: text
  - name: companyDomain
    display_name: Company Domain
    help_text: Clearbit identified domain for the company
    input_type: text
  ```
And so on...let me know if you need more clarification.

File: integration.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  products:
+  - ab_testing
+  - personalization
+  description: Use this integration to create Optimizely
+    audiences based on visitor data from Clearbit. This
+    integration requires an existing account with Clearbit.
+  developer: Clearbit
+  developer_website: https://clearbit.com
+  enabled: true
+  logo_file_name: logo.png
```
Comment (aliabbasrizvi) [inline integration.yaml]: nit. I would just call the file `clearbit.png` and rename it accordingly under `assets`

File: integration.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  master_label: Clearbit
+  partner_dir_url: https://www.optimizely.com/partners/technology/clearbit
```
Comment (aliabbasrizvi) [inline integration.yaml]: @lucasoptimizely FYI

File: integration.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  permission_required: clearbit_integration
+  settings_metadata:
+    fields:
+    - inputType: text
```
Comment (aliabbasrizvi) [inline integration.yaml]: If this needs to be hidden in the UI and in the API response then I would recommend that you change inputType to `password`

**Comment by davidlumley** [comment]
@maccman Should we merge or close this?

**Comment by maccman** [comment]
So sad

**Pull request closed without merging**